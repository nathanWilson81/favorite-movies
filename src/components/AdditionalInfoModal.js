import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { noop } from 'lodash/fp'

const Backdrop = styled.div`
  position: fixed;
  z-index: 200;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: rgba(15, 17, 20, 0.5);
`

const Content = styled.div`
  position: relative;
  z-index: 201;
  height: 60%;
  width: 60%;
  background: #fff;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 3rem;
`

const StyledTextarea = styled.textarea`
  display: block;
  min-width: 100%;
  max-width: 100%;
  resize: vertical;
  border: 3px solid lightgrey;
`

const onBackdropClick = (e, cb) => (e.target === e.currentTarget ? cb() : noop)

export const AdditionalInfoModal = ({
  isOpen,
  onClose,
  updateMovie,
  movie
}) => (
  <React.Fragment>
    {isOpen && (
      <Backdrop onClick={e => onBackdropClick(e, onClose)}>
        <Content>
          <h1>Additional Information</h1>
          <StyledTextarea
            placeholder="Ex. Pros / Cons, Director, IMDB link"
            rows={10}
            value={movie.additionalInformation}
            onChange={e =>
              updateMovie(movie, { additionalInformation: e.target.value })
            }
          />
        </Content>
      </Backdrop>
    )}
  </React.Fragment>
)

AdditionalInfoModal.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
  updateMovie: PropTypes.func.isRequired,
  movie: PropTypes.object
}

AdditionalInfoModal.defaultProps = {
  movie: null
}

export default AdditionalInfoModal
