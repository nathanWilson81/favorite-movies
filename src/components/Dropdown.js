import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { map } from 'lodash/fp'

const StyledSelect = styled.select`
  height: 22px;
  background: #fff;
`

export const Dropdown = ({ onChange, value, values }) => (
  <StyledSelect onChange={onChange} value={value}>
    {map(
      val => (
        <option key={val.value} value={val.value}>
          {val.label}
        </option>
      ),
      values
    )}
  </StyledSelect>
)

Dropdown.propTypes = {
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
  values: PropTypes.array.isRequired
}

export default Dropdown
