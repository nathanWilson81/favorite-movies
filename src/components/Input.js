import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const StyledInput = styled.input`
  width: 350px;
  border: none;
  height: 1.5rem;
`

export const Input = ({ onChange, value, placeholder }) => (
  <StyledInput
    type="text"
    onChange={onChange}
    value={value}
    placeholder={placeholder}
  />
)

Input.propTypes = {
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
  placeholder: PropTypes.string
}

Input.defaultProps = {
  placeholder: ''
}

export default Input
