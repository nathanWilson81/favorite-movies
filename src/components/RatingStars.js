import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import MaterialIcon from 'material-icons-react'

// !important hack to get around material-icons bug where changing props wasn't changing the color of the icon
const ClickableIcon = styled.span`
  margin-top: 5px;
  cursor: pointer;
  i {
    color: ${p => (p.active ? 'gold !important' : '')};
  }
`

const StarsContainer = styled.div`
  display: flex;
`

export const RatingStars = ({ starCount, updateMovie, movie }) => {
  const stars = []
  for (let i = 1; i < starCount + 1; i++) {
    stars.push(
      <ClickableIcon
        key={i}
        active={movie.movieRating >= i}
        onClick={() => updateMovie(movie, { movieRating: i })}
      >
        <MaterialIcon icon="grade" />
      </ClickableIcon>
    )
  }
  return <StarsContainer>{stars}</StarsContainer>
}

RatingStars.propTypes = {
  starCount: PropTypes.number.isRequired,
  updateMovie: PropTypes.func.isRequired,
  movie: PropTypes.object.isRequired
}

export default RatingStars
