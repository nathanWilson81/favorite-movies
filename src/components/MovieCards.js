import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { map } from 'lodash/fp'
import MaterialIcon from 'material-icons-react'
import { Input, Dropdown, RatingStars } from '../components'

const Card = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-evenly;
  margin-bottom: 1rem;
  background: #fff;
  padding: 0.5rem;
  box-shadow: 3px 3px 3px rgba(0, 0, 0, 0.03);
  min-height: 45px;
  min-width: 450px;
`

const ClickableIcon = styled.span`
  margin-top: 5px;
  cursor: pointer;
`

export const MovieCards = ({
  movies,
  deleteMovie,
  updateMovie,
  openAdditionalInfoModal,
  starCount,
  genres
}) => (
  <React.Fragment>
    {map(
      movie => (
        <Card key={movie.id}>
          <ClickableIcon onClick={() => deleteMovie(movie)}>
            <MaterialIcon icon="delete" />
          </ClickableIcon>
          <Input
            type="text"
            onChange={e => {
              e.preventDefault()
              updateMovie(movie, { movieName: e.target.value })
            }}
            value={movie.movieName}
            placeholder="Enter your movie's name"
          />
          <Dropdown
            value={movie.movieGenre}
            onChange={e => updateMovie(movie, { movieGenre: e.target.value })}
            values={genres}
          />
          <RatingStars
            starCount={starCount}
            updateMovie={updateMovie}
            movie={movie}
          />
          <ClickableIcon onClick={() => openAdditionalInfoModal(movie)}>
            <MaterialIcon icon="info" />
          </ClickableIcon>
        </Card>
      ),
      movies
    )}
  </React.Fragment>
)

MovieCards.propTypes = {
  movies: PropTypes.array.isRequired,
  deleteMovie: PropTypes.func.isRequired,
  updateMovie: PropTypes.func.isRequired,
  openAdditionalInfoModal: PropTypes.func.isRequired,
  starCount: PropTypes.number.isRequired,
  genres: PropTypes.array.isRequired
}

export default MovieCards
