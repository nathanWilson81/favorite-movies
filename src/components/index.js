import { Dropdown } from './Dropdown'
import { Input } from './Input'
import { RatingStars } from './RatingStars'
import { MovieCards } from './MovieCards'
import { AdditionalInfoModal } from './AdditionalInfoModal'

export { Dropdown, Input, RatingStars, MovieCards, AdditionalInfoModal }
