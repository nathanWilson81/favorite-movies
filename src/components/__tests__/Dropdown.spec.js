import React from 'react'
import renderer from 'react-test-renderer'
import { Dropdown } from '../../components'

describe('Dropdown', () => {
  it('should match a snapshot', () => {
    const component = renderer.create(
      <Dropdown
        onChange={jest.fn()}
        value={'test'}
        values={[{ label: 'test', value: 'test' }]}
      />
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
