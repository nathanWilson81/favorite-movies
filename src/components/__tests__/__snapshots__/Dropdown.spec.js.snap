// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should match a snapshot 1`] = `
<select
  className="sc-bdVaJa fqmIub"
  onChange={[MockFunction]}
  value="test"
>
  <option
    value="test"
  >
    test
  </option>
</select>
`;
